src/Directory.Build.targets

```diff
@@ -118,6 +118,9 @@
     </ItemGroup>
   </Target>
 
+  <!-- This target creates a type library for the assemblies in a package. It requires desktop msbuild to locate the tooling
+       Type libraries are still needed by Visual Studio in some cases, like the legacy C# project system.
+       It calls into msbuild via COM, and needs the type library to do so. -->
   <Target Name="CreateTypeLib" BeforeTargets="AfterBuild" Inputs="$(TargetPath)" Outputs="$(TargetDir)$(TargetName).tlb;$(TargetDir)x64\$(TargetName).tlb" Condition="'$(BuildingInsideVisualStudio)' != 'true' and '$(CreateTlb)' == 'true' and $([MSBuild]::IsOSPlatform('windows')) and '$(TargetFrameworkIdentifier)' == '.NETFramework' and '$(MSBuildRuntimeType)' != 'Core'">
```

Comment:
I saw you disabled this explicitly--was the MSBuildRuntimeType check here insufficient?

Reply:
You mean disabling via CreateTlb? There is use elsewhere in src/MSBuild/MSBuild.csproj https://github.com/dotnet/msbuild/pull/9685/files#diff-24f179152546e553997c333eeb9c000c2e31d5b3ff953b524fc49cdfc781fb6dR242

I thought it was clearer to have an explicit disabling of the creation rather than an implicit one based on the build host.